scalableminds/webknossos
# Tree search

### Mailable description of changes:
 - Trees can be searched via a search menu within the trees tab on the right hand side. In contrast to the browser search functionality (CTRL + F) this search will consider **all** trees regardless of whether they are visible in the viewport.
- The comments tab will load significantly faster for a large amount of trees.

![image](https://user-images.githubusercontent.com/2486553/40543358-42ea6e6e-6024-11e8-85fa-c870d04a674c.png)


### Steps to test:
- Open a tracing with trees and use the search functionality in the tree tabs view
- Clicking a selected tree (or hitting enter) should close the search and focus the tree in the list.

### Notes
I tested the search feature for a tracing with 10 000 trees and the performance was near-instant. While testing that I noticed that the comments tab took 7 seconds to load. I fixed that by pulling a filter in front of a sort, since I measured that the sorting was the bottleneck. Now, the comment tab performance loads near-instant (for zero comments).

### Issues:
- fixes #2632 

------
